src/Models/DomesticAddress.php

```diff
@@ -4,29 +4,72 @@
 
 namespace Tipoff\Addresses\Models;
 
+use Assert\Assert;
+use Carbon\Carbon;
 use Tipoff\Support\Models\BaseModel;
 use Tipoff\Support\Traits\HasPackageFactory;
 
+/**
+ * @property int id
+ * @property string address_line_1
+ * @property string address_line_2
+ * @property City city
+ * @property Zip zip
+ * @property Carbon created_at
+ * @property Carbon updated_at
+ * // Raw relations
+ * @property int city_id
+ * @property string zip_code
+ */
 class DomesticAddress extends BaseModel
 {
     use HasPackageFactory;
 
-    protected $casts = [];
+    protected $casts = [
+        'id' => 'integer',
+        'city_id' => 'integer',
+    ];
+
+    protected $fillable = [
+        'address_line_1',
+        'address_line_2',
+        'city_id',
+        'zip_code',
+    ];
+
+    /**
+     * @param string $line1
+     * @param string|null $line2
+     * @param string|City $city
+     * @param string|Zip $zip
+     * @return static
+     */
+    public static function createDomesticAddress(string $line1, ?string $line2, $city, $zip): self
+    {
+        $zip = ($zip instanceof Zip) ? $zip : Zip::query()->findOrFail(trim($zip));
+        $city = ($city instanceof City) ? $city : City::query()->byTitle($city)->firstOrFail();
```

Comment:
Zip lookup makes sense as findOrFail.  I did city as find or fail as well.  Not sure if cities should be added automatically if not found, so flagging this decision.

Reply:
I think it's fine to add the City to that database if the title of the city isn't there. These won't be user created since we'll be implementing a single address field with the Google Places Autocomplete (#30) everywhere a User is giving us their address in the application.